src/sentry/api/decorators.py

```diff
@@ -38,8 +40,21 @@ def wrapped(self, request: Request, *args, **kwargs) -> Response:
 def email_verification_required(func):
     @wraps(func)
     def wrapped(self, request: Request, *args, **kwargs) -> Response:
-        if not request.user.get_verified_emails().exists():
-            raise EmailVerificationRequired(request.user)
+        if isinstance(request.user, User):
+            if not request.user.get_verified_emails().exists():
+                raise EmailVerificationRequired(request.user)
+        elif isinstance(request.user, APIUser):
+            has_verified_email = False
+            for email in request.user.useremails:
+                if email.is_verified:
+                    has_verified_email = True
+                    break
+            if not has_verified_email:
+                raise EmailVerificationRequired(request.user)
```

Comment:
Should this be a method on the APIUser? Similar to the one that exists on `User`?